### DISCUSSION TITLE
How to get incremental changes when subscribing to table

### DISCUSSION THREAD

**Discussion by whogotpwned** [enisdenjo/graphql-ws#517] [Q&A] [answered]
I am sifting through the documentation but can not find anything about how to get incremental changes.

This is how I instantiate my client:
```
  const client = createClient({
    url: <myURL>,
    connectionParams: () => {
      const token = <myAccessToken>;
      return {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      };
    },
  });
```

This is how I subscribe:

```
  const subscription = client.subscribe({
    query: <myQuery>,
    variables: {
      user_id: <first var>,
      contact: <second var>
    }
  }, {
    next(data) {
      console.log(data);

    },
    complete() {
      console.log('done');
    },
    error(e) {
      console.log(e);
    },
  });
``` 

And this is myQuery:

```
export const myQuery = gql`
    subscription myQuery($user_id: uuid!, $contact: uuid!){
        chats(where: {
            user_id: {
                _eq: $user_id
            },
            contact: {
                _eq: $contact
            }}) {
            id
        }
    }`
``` 

This actually works but it gives me the whole table when something changes but actually this is a total overload for me. I just want the changes so if a row was added it should give me the row and if something was deleted I want the data that has been deleted. Is there something I overlooked?

**Comment by enisdenjo** [comment, marked as answer]
This is more of a question of GraphQL than it is of transport. `graphql-ws` is just a transport library, allowing you to get realtime updates from GraphQL.

You have to visit your subscription implementation in GraphQL and investigate how to refactor it to stream incremental updates instead.
